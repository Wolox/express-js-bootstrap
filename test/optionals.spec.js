const utils = require('./helpers/utils'),
  { mockCommand } = require('./helpers/mocks'),
  { examplePrompts } = require('./helpers/constants');

beforeAll(() => mockCommand());

const optionals = [
  ['rollbar', ['package.json', 'app.js']],
  ['cors', ['package.json', 'app.js']],
  ['coveralls', ['package.json']]
];

describe.each(optionals)('Project with %s', (optionalFeature, files) => {
  beforeAll(() =>
    utils.runKickoff({
      ...examplePrompts,
      projectName: 'OptionalProject',
      optionalsFeatures: { [optionalFeature]: true }
    })
  );

  test.each(files)('creates expected %s', file => {
    expect(utils.getFileContent(`OptionalProject/${file}`)).toMatchSnapshot();
  });
});
