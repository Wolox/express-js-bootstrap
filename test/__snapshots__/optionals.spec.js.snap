// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project with cors creates expected app.js 1`] = `
"const express = require('express'),
bodyParser = require('body-parser'),
morgan = require('morgan'),
path = require('path'),
cors = require('cors'),
config = require('./config'),
routes = require('./app/routes'),
errors = require('./app/middlewares/errors'),
DEFAULT_BODY_SIZE_LIMIT = 1024 * 1024 * 10,
DEFAULT_PARAMETER_LIMIT = 10000;

const bodyParserJsonConfig = () => ({
parameterLimit: config.common.api.parameterLimit || DEFAULT_PARAMETER_LIMIT,
limit: config.common.api.bodySizeLimit || DEFAULT_BODY_SIZE_LIMIT
});

const bodyParserUrlencodedConfig = () => ({
extended: true,
parameterLimit: config.common.api.parameterLimit || DEFAULT_PARAMETER_LIMIT,
limit: config.common.api.bodySizeLimit || DEFAULT_BODY_SIZE_LIMIT
});

const app = express();

app.use(cors());

app.use('/docs', express.static(path.join(__dirname, 'docs')));

// Client must send \\"Content-Type: application/json\\" header
app.use(bodyParser.json(bodyParserJsonConfig()));
app.use(bodyParser.urlencoded(bodyParserUrlencodedConfig()));

if (!config.isTesting) {
morgan.token('req-params', req => req.params);
app.use(
  morgan(
    '[:date[clf]] :remote-addr - Request \\":method :url\\" with params: :req-params. Response status: :status.'
  )
);
}

routes.init(app);

app.use(errors.handle);

module.exports = app;
"
`;

exports[`Project with cors creates expected package.json 1`] = `
"{
  \\"name\\": \\"OptionalProject\\",
  \\"version\\": \\"0.1.0\\",
  \\"description\\": \\"Example\\",
  \\"engines\\": {
    \\"node\\": \\"8.9.12\\",
    \\"npm\\": \\"6.4.1\\"
  },
  \\"scripts\\": {
    \\"console\\": \\"node console.js\\",
    \\"cover\\": \\"NODE_ENV=testing jest --coverage --runInBand --forceExit test/app\\",
    \\"test\\": \\"NODE_ENV=testing jest test/app --runInBand --forceExit\\",
    \\"test-inspect\\": \\"NODE_ENV=testing node --inspect --debug-brk jest test/app.spec.js\\",
    \\"eslint-check\\": \\"eslint --print-config .eslintrc.js | eslint-config-prettier-check\\",
    \\"lint\\": \\"eslint \\\\\\"**/*.js\\\\\\"\\",
    \\"lint-diff\\": \\"git diff --name-only --cached --relative | grep \\\\\\\\\\\\\\\\.js$ | xargs eslint\\",
    \\"lint-fix\\": \\"eslint \\\\\\"**/*.js\\\\\\" --fix\\",
    \\"precommit\\": \\"npm run lint-diff\\",
    \\"outdated\\": \\"npm outdated --depth 0\\",
    \\"pretest\\": \\"npm run lint\\",
    \\"prestart\\": \\"npm run lint\\",
    \\"start\\": \\"nodemon --inspect server.js\\"
  },
  \\"cacheDirectories\\": [
    \\"node_modules\\"
  ],
  \\"main\\": \\"app.js\\",
  \\"author\\": \\"Wolox\\",
  \\"homepage\\": \\"https://test.com.ar\\",
  \\"license\\": \\"MIT\\",
  \\"repository\\": {
    \\"type\\": \\"git\\",
    \\"url\\": \\"https://test.com.ar.git\\"
  },
  \\"bugs\\": {
    \\"url\\": \\"https://test.com.ar/issues\\",
    \\"email\\": \\"tls@wolox.com.ar\\"
  },
  \\"jest\\": {
    \\"collectCoverageFrom\\": [
      \\"**/*.js\\",
      \\"!**/console.js\\",
      \\"!**/node_modules/**\\",
      \\"!**/build/**\\",
      \\"!**/migrations/**\\",
      \\"!**/config/**\\",
      \\"!**/scripts/**\\",
      \\"!**/docs/**\\"
    ],
    \\"testEnvironment\\": \\"node\\",
    \\"transform\\": {
      \\"^.+\\\\\\\\.js$\\": \\"babel-jest\\"
    }
  },
  \\"dependencies\\": {
    \\"bcryptjs\\": \\"^2.4.3\\",
    \\"body-parser\\": \\"^1.18.2\\",
    \\"cors\\": \\"^2.8.4\\",
    \\"express\\": \\"^4.16.2\\",
    \\"jwt-simple\\": \\"^0.5.1\\",
    \\"morgan\\": \\"^1.9.1\\",
    \\"umzug\\": \\"^2.1.0\\",
    \\"winston\\": \\"^2.4.0\\",
    \\"winston-daily-rotate-file\\": \\"^1.7.2\\"
  },
  \\"devDependencies\\": {
    \\"babel\\": \\"6.23.0\\",
    \\"babel-core\\": \\"6.26.0\\",
    \\"babel-eslint\\": \\"^8.2.2\\",
    \\"babel-jest\\": \\"^23.6.0\\",
    \\"jest\\": \\"^24.0.0\\",
    \\"supertest\\": \\"^3.4.2\\",
    \\"babel-preset-es2015\\": \\"6.24.1\\",
    \\"dictum.js\\": \\"^1.0.0\\",
    \\"dotenv\\": \\"^5.0.0\\",
    \\"eslint\\": \\"^4.8.0\\",
    \\"eslint-config-airbnb-base\\": \\"^12.0.2\\",
    \\"eslint-config-prettier\\": \\"^2.3.1\\",
    \\"eslint-plugin-import\\": \\"^2.6.1\\",
    \\"eslint-plugin-prettier\\": \\"^2.1.1\\",
    \\"husky\\": \\"^0.14.3\\",
    \\"istanbul\\": \\"^0.4.3\\",
    \\"mocha\\": \\"^5.0.1\\",
    \\"mocha-lcov-reporter\\": \\"^1.2.0\\",
    \\"nodemon\\": \\"^1.18.7\\",
    \\"prettier\\": \\"^1.8.2\\",
    \\"prettier-eslint\\": \\"^8.2.1\\",
    \\"prompt\\": \\"^1.0.0\\"
  }
}
"
`;

exports[`Project with coveralls creates expected package.json 1`] = `
"{
  \\"name\\": \\"OptionalProject\\",
  \\"version\\": \\"0.1.0\\",
  \\"description\\": \\"Example\\",
  \\"engines\\": {
    \\"node\\": \\"8.9.12\\",
    \\"npm\\": \\"6.4.1\\"
  },
  \\"scripts\\": {
    \\"console\\": \\"node console.js\\",
    \\"cover\\": \\"NODE_ENV=testing jest --coverage --runInBand --forceExit test/app\\",
    \\"test\\": \\"NODE_ENV=testing jest test/app --runInBand --forceExit\\",
    \\"test-inspect\\": \\"NODE_ENV=testing node --inspect --debug-brk jest test/app.spec.js\\",
    \\"coveralls\\": \\"npm run cover -- --report lcovonly && cat ./coverage/lcov.info | coveralls\\",
    \\"eslint-check\\": \\"eslint --print-config .eslintrc.js | eslint-config-prettier-check\\",
    \\"lint\\": \\"eslint \\\\\\"**/*.js\\\\\\"\\",
    \\"lint-diff\\": \\"git diff --name-only --cached --relative | grep \\\\\\\\\\\\\\\\.js$ | xargs eslint\\",
    \\"lint-fix\\": \\"eslint \\\\\\"**/*.js\\\\\\" --fix\\",
    \\"precommit\\": \\"npm run lint-diff\\",
    \\"outdated\\": \\"npm outdated --depth 0\\",
    \\"pretest\\": \\"npm run lint\\",
    \\"prestart\\": \\"npm run lint\\",
    \\"start\\": \\"nodemon --inspect server.js\\"
  },
  \\"cacheDirectories\\": [
    \\"node_modules\\"
  ],
  \\"main\\": \\"app.js\\",
  \\"author\\": \\"Wolox\\",
  \\"homepage\\": \\"https://test.com.ar\\",
  \\"license\\": \\"MIT\\",
  \\"repository\\": {
    \\"type\\": \\"git\\",
    \\"url\\": \\"https://test.com.ar.git\\"
  },
  \\"bugs\\": {
    \\"url\\": \\"https://test.com.ar/issues\\",
    \\"email\\": \\"tls@wolox.com.ar\\"
  },
  \\"jest\\": {
    \\"collectCoverageFrom\\": [
      \\"**/*.js\\",
      \\"!**/console.js\\",
      \\"!**/node_modules/**\\",
      \\"!**/build/**\\",
      \\"!**/migrations/**\\",
      \\"!**/config/**\\",
      \\"!**/scripts/**\\",
      \\"!**/docs/**\\"
    ],
    \\"testEnvironment\\": \\"node\\",
    \\"transform\\": {
      \\"^.+\\\\\\\\.js$\\": \\"babel-jest\\"
    }
  },
  \\"dependencies\\": {
    \\"bcryptjs\\": \\"^2.4.3\\",
    \\"body-parser\\": \\"^1.18.2\\",
    \\"express\\": \\"^4.16.2\\",
    \\"jwt-simple\\": \\"^0.5.1\\",
    \\"morgan\\": \\"^1.9.1\\",
    \\"umzug\\": \\"^2.1.0\\",
    \\"winston\\": \\"^2.4.0\\",
    \\"winston-daily-rotate-file\\": \\"^1.7.2\\"
  },
  \\"devDependencies\\": {
    \\"babel\\": \\"6.23.0\\",
    \\"babel-core\\": \\"6.26.0\\",
    \\"babel-eslint\\": \\"^8.2.2\\",
    \\"babel-jest\\": \\"^23.6.0\\",
    \\"jest\\": \\"^24.0.0\\",
    \\"supertest\\": \\"^3.4.2\\",
    \\"babel-preset-es2015\\": \\"6.24.1\\",
    \\"coveralls\\": \\"^3.0.0\\",
    \\"dictum.js\\": \\"^1.0.0\\",
    \\"dotenv\\": \\"^5.0.0\\",
    \\"eslint\\": \\"^4.8.0\\",
    \\"eslint-config-airbnb-base\\": \\"^12.0.2\\",
    \\"eslint-config-prettier\\": \\"^2.3.1\\",
    \\"eslint-plugin-import\\": \\"^2.6.1\\",
    \\"eslint-plugin-prettier\\": \\"^2.1.1\\",
    \\"husky\\": \\"^0.14.3\\",
    \\"istanbul\\": \\"^0.4.3\\",
    \\"mocha\\": \\"^5.0.1\\",
    \\"mocha-lcov-reporter\\": \\"^1.2.0\\",
    \\"nodemon\\": \\"^1.18.7\\",
    \\"prettier\\": \\"^1.8.2\\",
    \\"prettier-eslint\\": \\"^8.2.1\\",
    \\"prompt\\": \\"^1.0.0\\"
  }
}
"
`;

exports[`Project with rollbar creates expected app.js 1`] = `
"const express = require('express'),
bodyParser = require('body-parser'),
morgan = require('morgan'),
path = require('path'),
config = require('./config'),
routes = require('./app/routes'),
errors = require('./app/middlewares/errors'),
DEFAULT_BODY_SIZE_LIMIT = 1024 * 1024 * 10,
DEFAULT_PARAMETER_LIMIT = 10000;

const bodyParserJsonConfig = () => ({
parameterLimit: config.common.api.parameterLimit || DEFAULT_PARAMETER_LIMIT,
limit: config.common.api.bodySizeLimit || DEFAULT_BODY_SIZE_LIMIT
});

const bodyParserUrlencodedConfig = () => ({
extended: true,
parameterLimit: config.common.api.parameterLimit || DEFAULT_PARAMETER_LIMIT,
limit: config.common.api.bodySizeLimit || DEFAULT_BODY_SIZE_LIMIT
});

const app = express();


app.use('/docs', express.static(path.join(__dirname, 'docs')));

// Client must send \\"Content-Type: application/json\\" header
app.use(bodyParser.json(bodyParserJsonConfig()));
app.use(bodyParser.urlencoded(bodyParserUrlencodedConfig()));

if (!config.isTesting) {
morgan.token('req-params', req => req.params);
app.use(
  morgan(
    '[:date[clf]] :remote-addr - Request \\":method :url\\" with params: :req-params. Response status: :status.'
  )
);
}

routes.init(app);

app.use(errors.handle);

module.exports = app;
"
`;

exports[`Project with rollbar creates expected package.json 1`] = `
"{
  \\"name\\": \\"OptionalProject\\",
  \\"version\\": \\"0.1.0\\",
  \\"description\\": \\"Example\\",
  \\"engines\\": {
    \\"node\\": \\"8.9.12\\",
    \\"npm\\": \\"6.4.1\\"
  },
  \\"scripts\\": {
    \\"console\\": \\"node console.js\\",
    \\"cover\\": \\"NODE_ENV=testing jest --coverage --runInBand --forceExit test/app\\",
    \\"test\\": \\"NODE_ENV=testing jest test/app --runInBand --forceExit\\",
    \\"test-inspect\\": \\"NODE_ENV=testing node --inspect --debug-brk jest test/app.spec.js\\",
    \\"eslint-check\\": \\"eslint --print-config .eslintrc.js | eslint-config-prettier-check\\",
    \\"lint\\": \\"eslint \\\\\\"**/*.js\\\\\\"\\",
    \\"lint-diff\\": \\"git diff --name-only --cached --relative | grep \\\\\\\\\\\\\\\\.js$ | xargs eslint\\",
    \\"lint-fix\\": \\"eslint \\\\\\"**/*.js\\\\\\" --fix\\",
    \\"precommit\\": \\"npm run lint-diff\\",
    \\"outdated\\": \\"npm outdated --depth 0\\",
    \\"pretest\\": \\"npm run lint\\",
    \\"prestart\\": \\"npm run lint\\",
    \\"start\\": \\"nodemon --inspect server.js\\"
  },
  \\"cacheDirectories\\": [
    \\"node_modules\\"
  ],
  \\"main\\": \\"app.js\\",
  \\"author\\": \\"Wolox\\",
  \\"homepage\\": \\"https://test.com.ar\\",
  \\"license\\": \\"MIT\\",
  \\"repository\\": {
    \\"type\\": \\"git\\",
    \\"url\\": \\"https://test.com.ar.git\\"
  },
  \\"bugs\\": {
    \\"url\\": \\"https://test.com.ar/issues\\",
    \\"email\\": \\"tls@wolox.com.ar\\"
  },
  \\"jest\\": {
    \\"collectCoverageFrom\\": [
      \\"**/*.js\\",
      \\"!**/console.js\\",
      \\"!**/node_modules/**\\",
      \\"!**/build/**\\",
      \\"!**/migrations/**\\",
      \\"!**/config/**\\",
      \\"!**/scripts/**\\",
      \\"!**/docs/**\\"
    ],
    \\"testEnvironment\\": \\"node\\",
    \\"transform\\": {
      \\"^.+\\\\\\\\.js$\\": \\"babel-jest\\"
    }
  },
  \\"dependencies\\": {
    \\"bcryptjs\\": \\"^2.4.3\\",
    \\"body-parser\\": \\"^1.18.2\\",
    \\"express\\": \\"^4.16.2\\",
    \\"jwt-simple\\": \\"^0.5.1\\",
    \\"morgan\\": \\"^1.9.1\\",
    \\"rollbar\\": \\"^2.3.9\\",
    \\"umzug\\": \\"^2.1.0\\",
    \\"winston\\": \\"^2.4.0\\",
    \\"winston-daily-rotate-file\\": \\"^1.7.2\\"
  },
  \\"devDependencies\\": {
    \\"babel\\": \\"6.23.0\\",
    \\"babel-core\\": \\"6.26.0\\",
    \\"babel-eslint\\": \\"^8.2.2\\",
    \\"babel-jest\\": \\"^23.6.0\\",
    \\"jest\\": \\"^24.0.0\\",
    \\"supertest\\": \\"^3.4.2\\",
    \\"babel-preset-es2015\\": \\"6.24.1\\",
    \\"dictum.js\\": \\"^1.0.0\\",
    \\"dotenv\\": \\"^5.0.0\\",
    \\"eslint\\": \\"^4.8.0\\",
    \\"eslint-config-airbnb-base\\": \\"^12.0.2\\",
    \\"eslint-config-prettier\\": \\"^2.3.1\\",
    \\"eslint-plugin-import\\": \\"^2.6.1\\",
    \\"eslint-plugin-prettier\\": \\"^2.1.1\\",
    \\"husky\\": \\"^0.14.3\\",
    \\"istanbul\\": \\"^0.4.3\\",
    \\"mocha\\": \\"^5.0.1\\",
    \\"mocha-lcov-reporter\\": \\"^1.2.0\\",
    \\"nodemon\\": \\"^1.18.7\\",
    \\"prettier\\": \\"^1.8.2\\",
    \\"prettier-eslint\\": \\"^8.2.1\\",
    \\"prompt\\": \\"^1.0.0\\"
  }
}
"
`;
